src/offcanvas/offcanvas-config.ts

```diff
@@ -24,10 +24,11 @@ export interface NgbOffcanvasOptions {
 
   /**
    * If `true`, the backdrop element will be created for a given offcanvas.
+   * If 'static', clicking the backdrop won't close the offcanvas.
    *
    * Default value is `true`.
    */
-  backdrop?: boolean;
+  backdrop?: boolean | 'static';
```

Comment:
You should type `backdrop: boolean | 'static' = true` in `NgbOffcanvasConfig` too (below in this file).
At the moment it is `backdrop = true` and typescript is unhappy when one tries to set it to 'static':

![Screenshot 2022-08-02 at 16 01 26](https://user-images.githubusercontent.com/8074436/182394077-e277f85e-9188-4859-9f63-a69fa071a9d3.jpg)